### DISCUSSION TITLE
Temp File Deduplication Storage Location

### DISCUSSION THREAD

**Discussion by miltonh26** [borgbase/vorta#1340] [General]
As I understand it, Vorta (borg) creates a copy of backed up files in another location defined by where borg is located so deduplication can be accomplished. This causes a major issue. The amount of free disk space on the partition with borg must be equal or larger than the size of the files being backed up after compression. If I have this wrong, then please let me know.

Because my partition was running out of space, Vorta would error out, then fail, leaving temporary files behind. When I tried to clean them up I found it difficult to do. It seemed that  Vorta cleaned up the files itself. But another time it did not leaving about 10Gb of space unavailable. I am unable to find where the files are located so they could be manually deleted. Removing Vorta did not solve this. This brings me to my question:

Where are borg temp files located? 

Next question, why are 2 copies of all files necessary? Couldn't this be accomplished with a hash? Rather than store 2 copies of the files backed up for deduplication, would it be possible to have borg calculate a hash of each backed up file and then when a backup is executed again, compare the previously calculated hash against the file to determine if the file has changed. If it has proceed with an incremental backup and update the hash, otherwise ignore it and move on. Would this work?

Manjaro KDE Plasma 5.24.5
Kernal 5.13.19-2-MANJAO
Vorta v0.8.6 Flatpak 

Thanks.

**Comment by m3nu** [comment]
Vorta only has its database. Borg has some cache files to keep a local index. They are a few GB for very large repos max.

There is no copy of your files or any "temp files". Maybe your repo is local and you're putting your files there?

**Reply by real-yfprojects** [reply]
The files are backed up to a repository you specified in vorta. This can be a local folder on any partition or a remote server. [Introduction to borg](https://borgbackup.readthedocs.io/en/stable/quickstart.html)

**Reply by miltonh26** [reply]
Thanks for the replies. 

I am only backing up to a single partition with one backup location. The repo is not local. 

Here is what I was referring to mentioned in the borg documentation:
[https://borgbackup.readthedocs.io/en/stable/faq.html#the-borg-cache-eats-way-too-much-disk-space-what-can-i-do](url)
![image](https://user-images.githubusercontent.com/5038455/170937347-9cc586c4-78d9-426b-a7cb-0559a2a42621.png)

There are further discussions referenced in the same screenshot which discuss using cache space. Which leads to this conversation: 
[https://github.com/borgbackup/borg/issues/4740](url)

Literally, I could watch borg eat up my disk space at the same time on both the device being backed up to and the cache partition. 

Any ideas on how to free up the "missing" space? The 10Gb which was lost and I cannot find to delete. How to find it? It's someplace...

Thanks.

**Reply by m3nu** [reply]
Since you already cite Borg docs and Borg issues, you may want to raise your issue there. I fail to see how this relates to Vorta.

**Reply by real-yfprojects** [reply]
> Any ideas on how to free up the "missing" space? The 10Gb which was lost and I cannot find to delete. How to find it? It's someplace...

You already mention borgbackup/borg#4740 and if that is your issue why not head over there? And if it is a different issue, you could create a new issue at the borg repository as *m3nu* suggested. Additionally there is already a work around explained in the screenshot you posted. Additionally you can read [this doc article](https://borgbackup.readthedocs.io/en/stable/internals.html) to find out more about storage quotas and locations.
